93 openxlsx2 mixed column error

Olivia-Box-Power opened the pull request.

zac-gleisner committed:
Fix to correctly format numbers and text in mixed columns

zac-gleisner committed:
Merge branch '70-openxlsx2' into 93-openxlsx2-mixed-column-error

zac-gleisner committed:
Remove utils package import as not required

zac-gleisner committed:
Extend fix for mixed columns to all tables, covering text in numeric columns in cover, contents, and notes tabs.
Check for multiple notes.
Fixed lints.

chcrowley commented on R/utils-workbook-style.R:
```diff
@@ -93,8 +93,14 @@
   # Some columns may contain numbers but have suppression text in them, e.g.
   # '[c]', which makes the column character class. Find the likely numeric cols.
-  cols_numeric <- suppressWarnings(lapply(table, as.numeric))  # coerce columns to numeric
+  cols_numeric <- lapply(table,
```
This chunk is repeated a couple of times - could be a function?

CHCRowley reviewed (commented):
This works great to fix the number stored as text bug for mixed columns with characters within [square brackets]. If there are mixed columns containing characters without square brackets (e.g. £ or a) the generate_workbook function fails. I think it should either: error with a helpful message to the user or generate the workbook with a message that this formatting likely goes against accessibility guidance. What do you think?

zac-gleisner committed:
Added functions for determining numeric, currency and mixed columns.
Added functionality to process currencies as numeric to 2 decimal places by default.

zac-gleisner committed:
Add scales to imports.
Fix lints.

zac-gleisner committed:
Fix check for table cells containing numbers incorrectly using .determine_currency.
Fix number formats being applied to wrong rows.

zac-gleisner committed:
Simplify application of custom formats to mixed columns.

zac-gleisner committed:
Change to only apply fix for mixed columns to columns which contain numeric/currency values or notes in separate cells. Cells with numeric/currency values and notes in the same cell do not meet the AF guidance so are not covered by this fix.

zac-gleisner committed:
Better handling of mixed columns in complex tables.
Fix lints.

zac-gleisner committed:
Fix for numbers already formatted with commas

zac-gleisner committed:
Updated NEWS.md

zac-gleisner committed:
Add logic for empty cells in columns which are otherwise valid currency/numeric to format as currency/numeric.

zac-gleisner committed:
Rework of functions determining cell and column types.
Improved regex for notes allowing spaces before and after notes.

zac-gleisner committed:
Fix to allow for negative numbers/currencies.

zac-gleisner committed:
Followup fix for negative numbers.

zac-gleisner committed:
Improved regex for negative numbers.

zac-gleisner committed:
Better handling of notes to ensure whitespace is preserved.

zac-gleisner committed:
Simplify processing of mixed columns:
All currency symbols removed and values processed as numeric.
Currency symbols passed to .style_table function and applied as number formats.
Improved regex detecting currencies to exclude
symbols part of notes inside square brackets.
Tidied up imported functions.

zac-gleisner committed:
Move all logic determine numeric and currency cells/columns to .insert_table, and pass results to .style_table to apply number formats.
Improved currency regex to identify currency symbols before or after numbers.
Added tests for new .determine_ functions.

zac-gleisner committed:
Fix currency symbols not being applied to correct cells in .style_table.

zac-gleisner committed:
Simplification of data processing avoiding repeated use of regex to identify data types.

zac-gleisner committed:
Added empty currency_formats for non-table worksheets

zac-gleisner committed:
Apply mixed column formatting to all sheet types.
Additional tests for internal data type functions.

zac-gleisner committed:
Improved regexes and removed redundant check on  sheet_stype in .insert_title

Olivia-Box-Power committed:
Code style changes

zac-gleisner committed:
Ensure currency columns converted to numeric data type once currency symbols are removed.
Updated regex for consistency.
Removed redundant code from currency functions.

Olivia-Box-Power committed:
Code style changes

Olivia-Box-Power committed:
Update .insert_table

Olivia-Box-Power committed:
Update .style_table

Olivia-Box-Power committed:
Fix devtools check warning

Olivia-Box-Power committed:
Update NEWS file

Olivia-Box-Power committed:
Comment out tests

Olivia-Box-Power committed:
Left align text columns

Olivia-Box-Power committed:
Tidy namespeace and documentation

Olivia-Box-Power committed:
Fix issue with minus sign in regex

Olivia-Box-Power committed:
Fix bug in .style_table when calculating the width of a column

Olivia-Box-Power committed:
Fix typo
